src/components/site-form.tsx

```diff
@@ -254,6 +278,50 @@ export const SiteForm = ( {
 					<span className="font-semibold">{ __( 'Site name' ) }</span>
 					<TextControlComponent onChange={ setSiteName } value={ siteName }></TextControlComponent>
 				</label>
+				{ setUseCustomDomain && setCustomDomain && (
+					<div className="flex items-center gap-2 mb-6">
+						<input
+							type="checkbox"
+							id="use-custom-domain"
+							checked={ useCustomDomain }
+							onChange={ ( e ) => {
+								setUseCustomDomain( e.target.checked );
+								setCustomDomain( generateDomainFromSiteName( siteName ) );
+							} }
+						/>
+						<label htmlFor="use-custom-domain">{ __( 'Use custom domain' ) }</label>
+						<span className="text-a8c-gray-50 text-xs ml-2">
+							{ __( 'This will require your password to authorize changes to the hosts file.' ) }
+						</span>
+					</div>
+				) }
+
+				{ useCustomDomain && setCustomDomain && (
+					<div className="flex flex-col gap-2 mb-6">
+						<label htmlFor="custom-domain" className="font-semibold">
+							{ __( 'Domain name' ) }
+						</label>
+						<TextControlComponent
+							id="custom-domain"
+							value={ customDomain ?? '' }
+							onChange={ setCustomDomain }
```

Comment:
Could we regenerate the proposed domain when the user changes the name, similar to what happens for the directory name field?

Reply:
I don't think we should be doing that if the user already set a custom domain but might be good to do if the user didn't actually set the custom domain explicitly.

Reply:
Correct, this is how site name and directory name relation currently works - directory name is autogenerated when user edits the site name, and stops when user picks the directory manually.